src/main/kotlin/org/eclipse/tractusx/bpdmcertificatemanagement/service/CertificateService.kt

```diff
@@ -56,4 +61,33 @@ class CertificateService(
         return certificateMapping.toCertificateDocumentResponseDto(result)
 
     }
+
+    fun getCertificatesByBpn(bpn: String, pageRequest: PageRequest): PageDto<CertificateDocumentResponseDto> {
+        if (bpn.isBlank())
+            throw IllegalArgumentException("Provided business partner number is null or empty")
+
+        return when {
+            bpn.startsWith("BPNL") -> {
+                val certificates = certificateRepository.findByBusinessPartnerNumber(bpn, pageRequest)
+                if (certificates.isEmpty) {
+                    throw CertificateNotExists("Legal Entity", bpn)
+                }
+                certificates.toPageDto(certificateMapping::toCertificateDocumentResponseDto)
+            }
+
+            bpn.startsWith("BPNS") -> {
```

Comment:
I would expect such logic when inserting documents. Now there is a situation in which I put in a document with a wrong business partner number and I can't search for it.

Reply:
Yeah, i am planning to add all validation including business partner number, site details also registration number into the new pull request. All validation will be conducted while creating Certificate entity together.